taggit/managers.py

```diff
@@ -215,18 +258,82 @@ def slugs(self):
         return self.get_queryset().values_list('slug', flat=True)
 
     @require_instance_manager
-    def set(self, *tags):
-        self.clear()
-        self.add(*tags)
+    def set(self, *tags, **kwargs):
```

Comment:
Implementation intended to match [`ManyRelatedManager.set()`](https://github.com/django/django/blob/ed0ff913c648b16c4471fc9a9441d1ee48cb5420/django/db/models/fields/related_descriptors.py#L907) as closely as possible without breaking the existing API.